linode_api4/objects/serializable.py

```diff
@@ -32,19 +38,20 @@ def from_json(cls, json: Dict[str, Any]) -> "JSONObject":
         for k in vars(obj):
             json_value = json.get(k)
             field_type = type_hints.get(k)
-
-            if json_value is None:
-                continue
-
-            if (
-                field_type is not None
-                and inspect.isclass(field_type)
-                and issubclass(field_type, JSONObject)
-            ):
-                setattr(obj, k, field_type.from_json(json_value))
-                continue
-
-            setattr(obj, k, json_value)
+            if list in (field_type, get_origin(field_type)):
+                type_hint_args = get_args(field_type)
+                if len(type_hint_args) > 0:
+                    list_item_type = type_hint_args[0]
+                    setattr(
+                        obj,
+                        k,
+                        [
+                            cls._try_from_json(item, list_item_type)
+                            for item in json_value
+                        ],
+                    )
+                    continue
+            setattr(obj, k, cls._try_from_json(json_value, field_type))
```

Comment:
Optional:
For readability you might be able to break this out into a few functions:

```python
    @classmethod
    def _parse_attr_list(cls, json_value, field_type):
        """
        Attempts to parse a list attribute with a given value and field type.
        """

        type_hint_args = get_args(field_type)

        if len(type_hint_args) < 1:
            return cls._try_from_json(json_value, field_type)

        return [
            cls._try_from_json(item, type_hint_args[0]) for item in json_value
        ]

    @classmethod
    def _parse_attr(cls, json_value, field_type):
        """
        Attempts to parse an attribute with a given value and field type.
        """

        if list in (field_type, get_origin(field_type)):
            return cls._parse_attr_list(json_value, field_type)

        return cls._try_from_json(json_value, field_type)

    @classmethod
    def from_json(cls, json: Dict[str, Any]) -> "JSONObject":
        """
        Creates an instance of this class from a JSON dict.
        """
        obj = cls()

        type_hints = get_type_hints(cls)

        for k in vars(obj):
            setattr(obj, k, cls._parse_attr(json.get(k), type_hints.get(k)))

        return obj
```

Reply:
I'll do that now, good idea!